mod/caching/set/type_plus_right/wikirate_analysis/source.rb

```diff
@@ -1,8 +1,18 @@
 include Card::CachedCount
 
 expired_cached_count_cards do |changed_card|
-  if (l=changed_card.left)  && l.type_code == :source &&
-     (r=changed_card.right) && r.type_code == :pointer && (r.key == 'company' || r.key == 'topic')
-    changed_card.item_cards
+  if (l = changed_card.left) && l.type_code == :source &&
+     (r = changed_card.right) && (r.key == 'company' || r.key == 'topic') &&
+     changed_card.type_code == :pointer
+    
+    # find all related analysis
+    card_names = changed_card.item_names.unshift('in')
+    analysis_type_id = Card::WikirateAnalysisID
+    case r.key
+    when 'company'
+      Card.search type_id: analysis_type_id, left: card_names, append: 'source'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.